Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,416 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+#ifndef M_PI
+    #define M_PI 3.14159265358979323846
+#endif
+
+float MiscWidth[30], MiscHeight[30];
+float RadarWidth[23], RadarHeight[23];
+float HUDWidth[39], HUDHeight[34];
+float CameraWidth[1];
+
+float                  ScreenWidthScale = 1.0f;
+static constexpr float ScreenHeightScale = 1.0f;
+
+static constexpr float HudWidthScale = 1.0f;
+static constexpr float HudHeightScale = 1.0f;
+static constexpr float RadarWidthScale = 1.0f;
+static constexpr float RadarHeightScale = 1.0f;
+/* NOTE: Changing hud and radar scales may misalign elements */
+
+constexpr float WidthMult = 0.0015625f;
+constexpr float HeightMult = 0.002232143f;
+
+float* ScreenAspectRatio;
+float* ScreenFieldOfView;
+
+float AdjustFOV(float f, float ar)
+{
+    return std::round((2.0f * atan(((ar) / (4.0f / 3.0f)) * tan(f / 2.0f * ((float)M_PI / 180.0f)))) * (180.0f / (float)M_PI) * 100.0f) / 100.0f;
+}
+
+void SetFOV(float factor)
+{
+    *ScreenFieldOfView = AdjustFOV(factor, *ScreenAspectRatio);
+}
+
+void __declspec(naked) CalculateAimingPoint()
+{
+    _asm
+    {
+        fstp    st
+        mov     edx, [ScreenAspectRatio]
+        fmul[edx]
+        mov     edi, [esp + 1Ch + 14h]
+        mov     edx, edi
+        mov     ebx, 5149A6h
+        jmp     ebx
```

Comment:
Replace these instructions with jmp

```cpp
static constexpr DWORD RETURN_CCamera_Find3rdPersonCamTargetVector = 0x5149A6;
```

```cpp
jmp RETURN_CCamera_Find3rdPersonCamTargetVector
```

Reply:
Additionally, are you sure that this hook works properly? It seems to me that the value at the top of the FPU register may be random here after doing fstp st, unless I missed something